src/test/scala/chiselTests/stage/ChiselMainSpec.scala

```diff
@@ -275,8 +283,70 @@ class ChiselMainSpec extends AnyFeatureSpec with GivenWhenThen with Matchers wit
         message = Seq(Right("Fatal errors during hardware elaboration")),
         stdout = Seq(
           Right(
-            "src/test/scala/chiselTests/stage/ChiselMainSpec.scala:43:6: Invalid bit range (3,-1)"
-          )
+            "src/test/scala/chiselTests/stage/ChiselMainSpec.scala:44:6: Invalid bit range (3,-1)"
+          ),
+          // This spacing is important to make sure the caret is correct
+          // Keep the source-root test below in sync
+          Right("     w(3, -1)"),
+          Right("      ^")
+        )
+      )
+    )
+  }
+  Feature("Builder.error errors with source info even with incorrect source-root") {
+    runStageExpectException(
+      ChiselMainExceptionTest[ChiselException](
+        args = Array("-X", "low", "--source-root", ".."),
+        generator = Some(classOf[BuilderErrorModule]),
+        message = Seq(Right("Fatal errors during hardware elaboration")),
+        stdout = Seq(
+          Right(
+            "src/test/scala/chiselTests/stage/ChiselMainSpec.scala:44:6: Invalid bit range (3,-1)"
+          ),
+          Left("w(3, -1)")
+        )
+      )
+    )
+  }
+  Feature("Builder.error errors accept multiple source-roots") {
+    runStageExpectException(
+      ChiselMainExceptionTest[ChiselException](
+        args = Array("-X", "low", "--source-root", ".", "--source-root", "src/test/resources/chisel3/sourceroot1"),
+        generator = Some(classOf[BuilderErrorModuleFakeSourceInfo]),
+        message = Seq(Right("Fatal errors during hardware elaboration")),
+        stdout = Seq(
+          Right(
+            "Foo:3:10: Invalid bit range (3,-1)"
+          ),
+          // This spacing is important to make sure the caret is correct
+          // Keep the source-root test below in sync
+          Right("I am the file in sourceroot1"),
+          Right("         ^")
+        )
+      )
+    )
+  }
+  Feature("Builder.error errors accept multiple source-roots and pick first match") {
+    runStageExpectException(
+      ChiselMainExceptionTest[ChiselException](
+        args = Array(
+          "-X",
+          "low",
+          "--source-root",
+          "src/test/resources/chisel3/sourceroot2",
+          "--source-root",
+          "src/test/resources/chisel3/sourceroot1"
+        ),
+        generator = Some(classOf[BuilderErrorModuleFakeSourceInfo]),
+        message = Seq(Right("Fatal errors during hardware elaboration")),
+        stdout = Seq(
+          Right(
+            "Foo:3:10: Invalid bit range (3,-1)"
+          ),
+          // This spacing is important to make sure the caret is correct
+          // Keep the source-root test below in sync
+          Right("I am the file in sourceroot2"),
+          Right("         ^")
```

Comment:
as I said, can't you write a test that will hit a firrtl -> .v error (an uninitialized Wire(Bool()) should do it) which goes through emitVerilog and reflects the error in the Exception message?

Reply:
I added the same suite of tests to circt ChiselStageSpec which also includes checking that the --source-root to --include-dir propagation is correct